src/services/dynamo-database-service.ts

```diff
@@ -100,15 +100,24 @@ export class DynamoDatabaseService {
         ':ttl': { N: ttl.toString() },
         ':false': { BOOL: false },
       },
-      ConditionExpression: 'attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false',
+      ConditionExpression:
+        'attribute_exists(pk) AND (attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false)',
     };
     const command = new UpdateItemCommand(commandInput);
-    const response = await this.dynamoClient.send(command);
-    if (!response) {
-      const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
+    try {
+      const response = await this.dynamoClient.send(command);
+      logger.info(`${LOGS_PREFIX_SENSITIVE_INFO} Account ${userId} marked as deleted`);
+      logAndPublishMetric(MetricNames.MARK_AS_DELETED_SUCCEEDED);
```

Comment:
if the response is undefined, we are still logging this metric but then we are also logging DB UPDATE ERROR... is that intentional? should we not `logAndPublishMetric(MetricNames.MARK_AS_DELETED_SUCCEEDED);` only when the operation has definitely succeeded ?

Reply:
also.. what is the response if the update is NOT made because of the condition expression? I do not think it will throw an error if the update is not done because the condition is not met.. so then we may still be logging this success metric when it is not really relevant?

Reply:
When I tested it, it didn't log the 'MARK_AS_DELETED_SUCCEEDED' when the user_id didn't exist. If it's undefined it should just log DB_UPDATE_ERROR.

Reply:
okay - but as it is written right now, if the an error is not thrown but the response is undefined (which I know it is not expected) then we would first log a success metric and then log a db update error inside the if statement block. I would move the success metric logging just before the response is returned and also add a return statement inside the if block to ensure that we do not log both success and failure metric in one flow